datumaro/components/extractor.py

```diff
@@ -170,22 +176,91 @@ def get(self, id, subset=None):
                 return item
         return None
 
+T = TypeVar('T')
+
+class _ImportFail(DatumaroError):
+    pass
+
+class ImportErrorPolicy:
+    def report_item_error(self, error: ItemImportError) -> Optional[NoReturn]:
+        """
+        Allows to report a problem with a dataset item.
+
+        This function must either call fail() or return. If this function
+        returns, the extractor must skip the item.
+        """
+        raise NotImplementedError
+
+    def report_annotation_error(self,
+            error: AnnotationImportError) -> Optional[NoReturn]:
+        """
+        Allows to report a problem with a dataset item annotation.
+
+        This function must either call fail() or return. If this function
+        returns, the extractor must skip the annotation.
+        """
+        raise NotImplementedError
+
+    def fail(self, error: Exception) -> NoReturn:
+        raise _ImportFail from error
+
+@define(eq=False)
+class ImportContext:
+    progress_reporter: Optional[ProgressReporter] = None
+    error_policy: Optional[ImportErrorPolicy] = None
```

Comment:
IMO, it's a bit weird that one of these is called "reporter" and the other is called "policy" when both are similar objects - abstract interfaces for reporting various things. Maybe renaming the latter to `error_reporter` would be better?

Reply:
I can't say that error policy is just for reporting, because it is supposed to define extractor reaction. Progress reported has a direct function of progress reporting.